android/src/main/java/com/pkmnapps/nearby_connections/NearbyConnectionsPlugin.java

```diff
@@ -50,12 +54,54 @@ public class NearbyConnectionsPlugin implements MethodCallHandler, FlutterPlugin
     private Activity activity;
     private static final String SERVICE_ID = "com.pkmnapps.nearby_connections";
     private static MethodChannel channel;
+    private static EventChannel eventChannel;
     private static PluginRegistry.Registrar pluginRegistrar;
-
+    private static NearbyConnectionsStreamHandler nearbyConnectionsStreamHandler;
+    private static ParcelFileDescriptor[] senderPayloadPipe = null;
+    private static OutputStream senderOutputStream = null;
+    private static InputStream senderInputStream = null;
+    private static Thread senderOutputStreamThread = null;
+    private volatile byte[] bytes = null;
+    private volatile boolean isPipeOpen = false;
     private NearbyConnectionsPlugin(Activity activity) {
         this.activity = activity;
     }
-    public NearbyConnectionsPlugin(){}
+    public NearbyConnectionsPlugin(){
+    }
+
+//    Stream Handler for event channel for receiver side
+    static class NearbyConnectionsStreamHandler implements EventChannel.StreamHandler {
+
+        public InputStream inputReceiverStream;
+        private EventChannel.EventSink eventSink;
+        @Override
+        public void onListen(Object arguments, EventChannel.EventSink events) {
+            if (events == null) return;
+            eventSink = events;
+        }
+
+        @Override
+        public void onCancel(Object arguments) {
+        }
+
+        public void initializeInputReceiverStream(InputStream inputStream) {
+            inputReceiverStream = inputStream;
+            byte[] bytes;
+            while (true) {
+                try {
+                    if (inputReceiverStream.available() <= 0) break;
```

Comment:
Its possible the while loop ends up being faster than the stream speed and we would end up no longer listening to the receiver stream.

I think we should follow the android docs here : https://developers.google.com/nearby/connections/android/exchange-data#stream:~:text=SystemClock.elapsedRealtime()%20%2D%20lastRead)%20%3E%3D%20READ_STREAM_IN_BG_TIMEOUT

Basically, a configured timeout value (READ_STREAM_IN_BG_TIMEOUT) after which the stream could be considered dead.

We should also let the user change the timeout (maybe a method channel call to update the stream timeout)

Reply:
Also on timeout we should call `events.endOfStream()`